pkg/sfu/downtrack.go

```diff
@@ -540,82 +544,27 @@ func (d *DownTrack) MaxTemporalLayer() int32 {
 	return d.maxTemporalLayer.get()
 }
 
-// switchSpatialLayer switches the current layer
-func (d *DownTrack) switchSpatialLayer(targetLayer int32) error {
-	if d.trackType != SimulcastDownTrack {
-		return ErrSpatialNotSupported
-	}
-
-	// already set
-	if d.CurrentSpatialLayer() == targetLayer {
-		return nil
-	}
-
+// switchSpatialLayer switches the target layer
+func (d *DownTrack) switchSpatialLayer(targetLayer int32) {
 	d.targetSpatialLayer.set(targetLayer)
-	return nil
 }
 
-func (d *DownTrack) UptrackLayersChange(availableLayers []uint16, layerAdded bool) (int32, error) {
```

Comment:
Forgot to clean this up in the last round. This was setting layers. Should not have. Everything should come from `streamallocator`.